model/file.go

```diff
@@ -22,6 +22,36 @@ type FileUploadResponse struct {
 	ClientIds []string    `json:"client_ids"`
 }
 
+const (
+	FILE_SORT_BY_CREATED      = "CreateAt"
+	FILE_SORT_BY_SIZE         = "Size"
+	FILE_SORT_BY_USERNAME     = "Username"
+	FILE_SORT_BY_CHANNEL_NAME = "Name"
+
+	FILE_SORT_ORDER_ASCENDING  = "ASC"
+	FILE_SORT_ORDER_DESCENDING = "DESC"
+)
+
+// GetFilesOptions contains options for getting files
+type GetFilesOptions struct {
+	// UserIds optionally limits the files to those created by the given users.
+	UserIds []string
+	// ChannelIds optionally limits the files to those created in the given channels.
+	ChannelIds []string
+	// Since optionally limits files to those created after a specified time as Unix time in milliseconds.
+	Since int64
+	// IncludeDeleted includes deleted files if set.
+	IncludeDeleted bool
+	// Page optionally limits to the requested page of results.
+	Page int
+	// PerPage optionally limits the number of results to fetch per page.
+	PerPage int
+	// SortBy optionally sorts the files by this field.
+	SortBy string
+	// SortDirection sorts the files in a particular order, default is ASCENDING if SortBy is provided.
```

Comment:
The constant `FILE_SORT_ORDER_ASCENDING` is just `ASC`, not `ASCENDING` which might be confusing. To my surprise, we don't have any convention in the code to unify this. Should we define a reusable Go type to enforce this?
```go
type SortDirection int
const (
    Ascending SortDirection = 0
    Descending SortDirection = 1
)
```

I pick `int` only because its default value is "ready to go" vs. having to handle the empty string. 0/5.

Reply:
I used strings here because the switch case in the query code becomes less nested because I don't have to set the sql string based on a number comparison.

Defining a type is good because it can enforce some compile time checks (and helps with code completion).

How about `byte` instead?

Reply:
I'd be inclined to stick with an `int` vs. a `byte` -- the latter is more compact, but typically also signals a different semantic meaning.

I do appreciate your point re: building the SQL query, but suspect that we can just wrap this with a helper method as needed. One potential thought to my original proposal is that it would be cool if `Ascending == -Descending`, for when you want to have a `sortDirection` and easily invert it. Maybe instead of the complexity of a type, we just have a `SortAscending bool` field, solving all the pain points?

Reply:
>...we just have a SortAscending bool field, solving all the pain points?

This leads to the "hidden" implication that if `SortAscending` is not set, then the results are ordered in descending order. I will add a note in the docs, so I'm fine with this, but it be a gotcha to people who dont read the docs since its no explicit enough.

Reply:
Oops, you're right: what about `SortDescending` instead?

Reply:
Sounds good to me.

I'll ping you once the PR is ready for a second round of reviews.